src/fmpz_mat/is_reduced.c

```diff
@@ -55,7 +55,12 @@ fmpz_mat_is_reduced(const fmpz_mat_t A, double delta, double eta)
 
         gr_ctx_clear(ctx);
 
-        if (is_reduced != T_UNKNOWN)
+        if (is_reduced != T_UNKNOWN || prec >= exact_cutoff)
+            break;
+        /* Failure with the fmpq context implied that we attempted to divide
+           by an exactly zero GS norm; by the current definition, this
+           matrix is not reduced. */
+        if (prec >= exact_cutoff)
```

Comment:
This second if statement will never evaluate to true, as the condition is already checked in the previous one. Is this intended?